test/powershell/Modules/Microsoft.PowerShell.Security/CmsMessage.Tests.ps1

```diff
@@ -0,0 +1,400 @@
+    }
+
+    It "Verify CMS message recipient resolution by cert" -Skip:(!$IsWindows) {
+        $ers = $null
+        $cert = Get-PfxCertificate $certLocation
+        $recipient = [System.Management.Automation.CmsMessageRecipient] $cert
+        $recipient.Resolve($ExecutionContext.SessionState, "Encryption", [ref] $ers)
+
+        $recipient.Certificates[0].Subject | Should Match 'CN=MyDataEnciphermentCert'
+    }
+
+    It "Verify a CMS message can be protected / unprotected" -Skip:(!$IsWindows) {
+        $protected = "Hello World","How are you?" | Protect-CmsMessage -To $certLocation
+        $protected.IndexOf("-----BEGIN CMS-----") | Should Be 0
+
+        $message = $protected | Get-CmsMessage
+        $message.Recipients[0].IssuerName | Should Be "CN=MyDataEnciphermentCert"
+
+        $decrypted = $message | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+
+        $decrypted = $protected | Unprotect-CmsMessage -To $certLocation
+        $decrypted | Should Be "Hello World`r`nHow are you?"
+    }
+}
+
+
+Describe "CmsMessage cmdlets thorough tests" -Tags "Feature" {
+
+    BeforeAll {
+        if ($IsWindows)
+        {
+            $certLocation = Create-TestCertificate
+            $certLocation | Should Not BeNullOrEmpty | Out-Null
+
+            if ($IsCoreCLR)
+            {
+                # PKI module is not available for PowerShell Core, so we need to use Windows PowerShell to import the cert
+                $fullPowerShell = Join-Path "$env:SystemRoot" "System32\WindowsPowerShell\v1.0\powershell.exe"
+
+                try {
+                    $modulePathCopy = $env:PSMODULEPATH
+                    $env:PSMODULEPATH = $null
+
+                    $importedCertPath = & $fullPowerShell -NoProfile -NonInteractive `
+                                                          -Command "Import-PfxCertificate $certLocation -CertStoreLocation cert:\CurrentUser\My | % PSPath"
+                    $importedCert = Get-ChildItem $importedCertPath
+                } finally {
+                    $env:PSMODULEPATH = $modulePathCopy
+                }
+            }
+            else
+            {
+                $importedCert = Import-PfxCertificate $certLocation -CertStoreLocation cert:\CurrentUser\My
+            }
+        }
+        else
+        {
+            # Skip for non-Windows platforms
+            $defaultParamValues = $PSdefaultParameterValues.Clone()
+            $PSdefaultParameterValues = @{ "it:skip" = $true }
+        }
+    }
+
+    AfterAll {
+        if ($defaultParamValues)
+        {
+            $PSdefaultParameterValues = $defaultParamValues
+        }
+
+        if ($importedCert)
+        {
+            Remove-Item (Join-Path Cert:\CurrentUser\My $importedCert.Thumbprint)
+        }
+    }
+
+    It "Verify message recipient resolution by Base64Cert" {
+        $certContent = "
```

Comment:
I would like to keep this Base64 encoded cert for
1. to use a Base64 encoded cert directly for `CmsMessageRecipient`, ascii armor `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` need to be present to surround the cert text.
2. this test is to verify if the passed-in text can be properly encoded (remove armor and then decode Base64), and thus it chooses `Encryption` purpose because that doesn't require the private key in the cert. That's why the cert text here is smaller in size.